libs/workspace/src/schematics/dt-component-example/index.ts

```diff
@@ -0,0 +1,307 @@
+ * @param importPath The path of the import.
+ * @param modulePath The path of module file to add the import into.
+ */
+function addImport(
+  sourceFile: ts.SourceFile,
+  name: string,
+  importPath: string,
+  modulePath: string,
+): InsertChange {
+  const lastImport = findNodes(
+    sourceFile,
+    ts.SyntaxKind.ImportDeclaration,
+  ).pop() as ts.ImportDeclaration;
+  // +1 because of new line
+  const end = lastImport.end + 1;
+  const toInsertImport = `import { ${name} } from './${importPath}';`;
+  return new InsertChange(modulePath, end, toInsertImport);
+}
+
+/**
+ * Add export statement to barrel files.
+ * @param sourceFile The source of the file.
+ * @param exportPath The path that we need to export.
+ * @param modulePath The path of the file.
+ */
+function addExport(
+  sourceFile: ts.SourceFile,
+  exportPath: string,
+  modulePath: string,
+): InsertChange {
+  const lastExport = findNodes(
+    sourceFile,
+    ts.SyntaxKind.ExportDeclaration,
+  ).pop() as ts.ExportDeclaration;
+  const end = lastExport.end + 1;
+  const toInsertExport = `export * from './${exportPath}';`;
+  return new InsertChange(modulePath, end, toInsertExport);
+}
+
+/**
+ * Function to update the module files.
+ * @param options
+ */
+function updateModules(options: DtExampleExtendedOptions): Rule {
+  return (host: Tree) => {
+    // add import
+    const modulePath = join(
+      'libs',
+      'examples',
+      'src',
+      options.dashName,
+      `${options.dashName}-examples.module.ts`,
+    );
+    const sourceFile = getSourceFile(host, modulePath);
+
+    const importChange = addImport(
+      sourceFile,
+      options.exampleComponent.component,
+      options.exampleRoute,
+      modulePath,
+    );
+
+    // find last module in the array, and add new module
+    const modulesDeclaration = findNodes(
+      sourceFile,
+      ts.SyntaxKind.VariableDeclaration,
+    ).find(
+      (node: ts.VariableDeclaration) =>
+        node.name.getText() === options.exampleComponent.modulesConstant,
+    ) as ts.VariableDeclaration;
+    const modulesElements = (modulesDeclaration.initializer as ts.ArrayLiteralExpression)
+      .elements;
+    const lastElement = modulesElements[modulesElements.length - 1];
+    const end = modulesElements.hasTrailingComma
+      ? lastElement.getEnd() + 1
+      : lastElement.getEnd();
+    const routesChange = new InsertChange(
+      modulePath,
+      end,
+      `${modulesElements.hasTrailingComma ? ' ' : ', '}${
+        options.exampleComponent.component
+      }`,
+    );
+
+    return commitChanges(host, [importChange, routesChange], modulePath);
+  };
+}
+
+/**
+ * Function to update the index file.
+ * @param options
+ */
+function updateIndex(options: DtExampleExtendedOptions): Rule {
+  return (host: Tree) => {
+    // add export to index.ts
+    const indexPath = join(
+      'libs',
+      'examples',
+      'src',
+      options.dashName,
+      'index.ts',
+    );
```

Comment:
NIT: Avoid splitting up paths as it would make searching and replacing for paths impossible for example you want to rename the path later on and you search for `libs/examples` this occurrence would not show up.

```suggestion
    const indexPath = join('libs/examples/src', options.dashName, 'index.ts');
```